src/erofs/format.rs

```diff
@@ -0,0 +1,279 @@
+use zerocopy::{
+    little_endian::{U16, U32, U64},
+    Immutable, IntoBytes, KnownLayout, TryFromBytes,
+};
+
+#[derive(Debug)]
+pub enum FormatError {
+    InvalidDataLayout,
+}
+
+pub const BLOCK_BITS: u8 = 12;
+pub const BLOCK_SIZE: usize = 1 << BLOCK_BITS;
+
+/* composefs Header */
+
+pub const COMPOSEFS_VERSION: U32 = U32::new(1);
+pub const COMPOSEFS_MAGIC: U32 = U32::new(0xd078629a);
+
+#[derive(Debug, Immutable, IntoBytes, TryFromBytes)]
+#[repr(u32)]
+pub enum ComposefsFlags {
+    HasAcl = 1 << 0,
+}
+
+#[derive(Debug, Default, Immutable, IntoBytes, KnownLayout, TryFromBytes)]
+#[repr(C)]
+pub struct ComposefsHeader {
+    pub magic: U32,
+    pub version: U32,
+    pub flags: U32,
+    pub composefs_version: U32,
+    pub unused: [U32; 4],
+}
+
+/* Superblock */
+
+pub const MAGIC_V1: U32 = U32::new(0xE0F5E1E2);
+pub const FEATURE_COMPAT_MTIME: U32 = U32::new(2);
+pub const FEATURE_COMPAT_XATTR_FILTER: U32 = U32::new(4);
+
+#[derive(Debug, Default, Immutable, IntoBytes, KnownLayout, TryFromBytes)]
+#[repr(C)]
+pub struct Superblock {
+    // vertical whitespace every 16 bytes (hexdump-friendly)
```

Comment:
This comment seems misplaced?

Reply:
It just means that there's an extra newline every time we hit a 16 byte boundary in the struct.  That makes it easier to see which part of the struct is which when looking at a hexdump.

It's all a bit redundant now, though, since we have a decent debugging tool.